### DISCUSSION TITLE
Question about file formats for genome alignment visualization in JBrowse

### DISCUSSION THREAD

**Discussion by VeselinaP** [GMOD/jbrowse-components#4656] [General] [answered]
Hello!
I use the JBrowse downloaded app on my computer to visualize genomes and map reads. I want to visualize 2 aligned genomes (each about 6 Mpb long). I did the alignment in Galaxy platform using the progressivemauve tool that generated an xmfa file (can convert to gff3). Can you please advice me on how I can view this alignment? I am not sure how to convert the xmfa file to suitable for JBrowse visualization format. 
I use windows 11 system that is very limited in bioinformatics. I tried to install pybigwig using python but it didn't worked out. 
Obviously I am not an bioinformatician and any help will be appreciated!

Thank you!

**Comment by cmdcolin** [comment]
This is a good question. I will do some brainstorming and get back to you!

**Reply by cmdcolin** [reply, marked as answer]
Hi there,
I looked into this a little and found that it might be awkward to make this available directly in jbrowse for the time being, but I tried making a xmfa to PAF converter tool, and PAF can be used in JBrowse. See tool here 

https://cmdcolin.github.io/xmfa2paf/


![image](https://github.com/user-attachments/assets/a88dac3f-2b52-4054-8df4-8cb5647e9dfd)

I was able to get two bacterial genomes aligned with Mauve to load in our synteny view

![image](https://github.com/user-attachments/assets/59022858-0600-4bc8-8532-403394803f0f)

**Reply by VeselinaP** [reply]
Hi, cmdcolin! Thanks a lot for the help! Yes it works, I was able to view the synteny track for my genomes. Do you think there is a way to view it as simple nucleotide alignment? I tried to zoom in but can't be seen as nucleotides.
Thanks a lot!

**Reply by cmdcolin** [reply]
you can load the alignment as a "normal track" but it doesn't support the X/= part of CIGAR strings well so it might display 'undefined' where there are SNPs

![image](https://github.com/user-attachments/assets/94482e90-d43b-43c4-b39d-5d5715f1a004)

that's kind of a bug though, we could try to find a workaround but it might take some time.


one way work around it outside of jbrowse would be to convert the PAF to BAM format (via SAM). there is a PAF to SAM script here, and then SAM can be converted to BAM https://gist.github.com/lh3/a1ef88085b2dc5454b694ccbf52630aa

the alternative is that the full synteny view does show the SNPs

![image](https://github.com/user-attachments/assets/29e783e6-b293-4137-b339-bfa915293289)

**Reply by cmdcolin** [reply]
(and my screenshot shows there might be an off by one on the SNPs there....will have to check if it's the xmfa to paf converter or jbrowse that is off by one)
